src/lib/RouterHistory.js

```diff
@@ -1,7 +1,43 @@
 // @flow
+import Router from './Router';
+
+export interface Route {
+  name: string;
+}
+
+export type RouterHistoryMethod = 'push' | 'replace';
+
+export type RouteParams = { [name: string]: any };
 
 export interface RouterHistory {
-  push(name: string, params?: { [name: string]: any }): void;
+  push(name: string, params?: RouteParams): void;
+  replace(name: string, params?: RouteParams): void;
+  getRoute(name: string): Route;
+  generate(name: string, params?: RouteParams): string;
```

Comment:
if this is generating a path name, I'd prefer `generatePathName` or `generatePath`

Reply:
Its generating a path name including query parameters. Will fix.